snuba/cli/cleanup.py

```diff
@@ -55,6 +50,19 @@ def cleanup(
     dataset = get_dataset(dataset_name)
     table = enforce_table_writer(dataset).get_schema().get_local_table_name()
 
-    clickhouse = ClickhousePool(clickhouse_host, clickhouse_port)
-    num_dropped = run_cleanup(clickhouse, database, table, dry_run=dry_run)
-    logger.info("Dropped %s partitions on %s" % (num_dropped, clickhouse_host))
+    if bool(clickhouse_host) ^ bool(clickhouse_port):
+        raise click.ClickException("Provide both Clickhouse host and port or neither")
+
+    # If Clickhouse host and port values are provided, run cleanup on that host,
+    # otherwise run cleanup on each registered cluster.
+    if clickhouse_host and clickhouse_port:
+        clickhouse_connections = [ClickhousePool(clickhouse_host, clickhouse_port)]
+    else:
+        clickhouse_connections = [
+            cluster.get_connection(ClickhouseClientSettings.READWRITE)
+            for cluster in CLUSTERS
+        ]
```

Comment:
This script uses the local table, so before we can use the Cluster configuration we need to figure out how to obtain a connection for the local table host since this will not be the same as what we have today in settings.CLICKHOUSE_HOST. 
So if someone runs this in prod with this change without passing the CLI parameters this will fail because it will try to cleanup the wrong host.

I would not apply this change to the scripts that act on local tables before we have a final decision on how to run these in prod with the new cluster system. I believe there are still open points there or am I forgetting about something ?

Reply:
Yeah, this part is only intended for use in the case where we do not have distributed tables right now. Distributed must always provide the host and port. (I'm not 100% sure if this is a use case that we want to support - we can either just make host and port required removing the default, or alternatively check for `local_dataset_mode` before using the one from the relevant cluster)

Reply:
We should try to avoid making this process different for the user between distributed and local tables, if possible at all. 
I am not familiar on how ops run this in prod. Do you have details? I think the question is whether breaking the abstraction that the cluster is fully configured through settings (and what clickhouse already knows) for this case is worth.

Reply:
Currently, prod passes the clickhouse host on the command line but uses the default port. (the default host wouldn't work since that's the distributed one).
Single tenant does not pass any value for host or port and therefore relies on the default being applied.
If we want to keep those scripts working it seems that we have to keep the different behaviour on local and distributed mode until the cluster can become aware of all the other nodes.

Reply:
This is updated in prod now